airflow/models/dag.py

```diff
@@ -1467,22 +1467,26 @@ def _test_cycle_helper(self, visit_map, task_id):
         """
         from airflow.models.dagbag import DagBag  # Avoid circular imports
 
-        # print('Inspecting %s' % task_id)
-        if visit_map[task_id] == DagBag.CYCLE_DONE:
-            return False
-
-        visit_map[task_id] = DagBag.CYCLE_IN_PROGRESS
-
-        task = self.task_dict[task_id]
-        for descendant_id in task.get_direct_relative_ids():
-            if visit_map[descendant_id] == DagBag.CYCLE_IN_PROGRESS:
-                msg = "Cycle detected in DAG. Faulty task: {0} to {1}".format(
-                    task_id, descendant_id)
+        path, visited = [task_id], []
+
+        while path:
+            current_task_id = path.pop()
+            if visit_map[current_task_id] == DagBag.CYCLE_DONE:
+                return False
```

Comment:
return False -> return

The return value does not seem to be checked and there is no 'return True'

Reply:
I think, we should add type hint for this method.